test/e2e/runSdkTests.mjs

```diff
@@ -109,24 +117,54 @@ async function runTest(dirName) {
     });
 
     const actorClient = client.actor(actor.id);
-    await actorClient.build('0.0', { waitForFinish: 30 * 60 * 1000 });
+    const build = await actorClient.build('0.0');
+    const buildResult = await client.build(build.id).waitForFinish({ waitSecs: 30 * 60 });
+    let exitCode = 0;
+
+    if (buildResult.status === 'SUCCEEDED') {
+        const testProcessFinished = Promise.withResolvers();
+        const testProcess = spawn(
+            process.argv[0],
+            [join(testDir, 'test.mjs'), actor.id],
+            {
+                stdio: 'pipe',
+                shell: false,
+            },
+        );
+
+        testProcess.stdout.on('data', (chunk) => process.stdout.write(chunk));
+        testProcess.stderr.on('data', (chunk) => process.stderr.write(chunk));
+
+        testProcess.on('exit', testProcessFinished.resolve);
 
-    const testProcessFinished = Promise.withResolvers();
-    const testProcess = spawn(
-        process.argv[0],
-        [join(testDir, 'test.mjs'), actor.id],
-        {stdio: 'inherit'},
-    );
+        await testProcessFinished.promise;
 
-    testProcess.on('close', testProcessFinished.resolve);
+        // Remove any monetization so that the Actor can be deleted
+        const { pricingInfos = [] } = await actorClient.get();
 
-    await testProcessFinished.promise;
+        if (pricingInfos.length > 0) {
+            await actorClient.update(
+                {
+                    pricingInfos: [
+                        ...pricingInfos,
+                        {
+                            pricingModel: 'FREE',
+                        },
+                    ],
+                },
+            );
+        }
+
+        exitCode = testProcess.exitCode;
+    } else {
```

Comment:
if you first check the inverse and throw, you can remove the nesting

```ts
if (buildResult.status !== 'SUCCEEDED') {
    throw new Error('Build failed');
}

// no nesting needed...
```

and the `exitCode` can be a const that way

Reply:
Not the biggest problem I had, but thanks for pointing it out! :grin: